clouddriver-aws/src/main/java/com/netflix/spinnaker/clouddriver/aws/deploy/ops/actions/UpdateAutoScalingGroup.java

```diff
@@ -60,19 +73,150 @@ public Result apply(
+                                  .getInstancesDistribution()
+                                  .getOnDemandPercentageAboveBaseCapacity())
+                      .withSpotAllocationStrategy(
+                          description.getSpotAllocationStrategy() != null
+                              ? description.getSpotAllocationStrategy()
+                              : existingMipInAsg
+                                  .getInstancesDistribution()
+                                  .getSpotAllocationStrategy())
+                      .withSpotInstancePools(
+                          description.getSpotInstancePools() != null
+                              ? description.getSpotInstancePools()
+                              : existingMipInAsg.getInstancesDistribution().getSpotInstancePools())
+                      .withSpotMaxPrice(
+                          StringUtils.isNotBlank(description.getSpotPrice())
+                              ? description.getSpotPrice()
+                              : existingMipInAsg.getInstancesDistribution().getSpotMaxPrice()));
+
+      updateReq.withMixedInstancesPolicy(mip);
+    } else {
+      if (command.isReqToUpgradeAsgToMixedInstancesPolicy) {
+        // create new MIP
+        MixedInstancesPolicy mip =
+            new MixedInstancesPolicy()
+                .withLaunchTemplate(
+                    new LaunchTemplate()
+                        .withLaunchTemplateSpecification(
+                            new LaunchTemplateSpecification()
+                                .withLaunchTemplateId(
+                                    command.launchTemplateVersion.getLaunchTemplateId())
+                                .withVersion(
+                                    String.valueOf(
+                                        command.launchTemplateVersion.getVersionNumber())))
+                        .withOverrides(command.launchTemplateOverrides))
+                .withInstancesDistribution(
+                    new InstancesDistribution()
+                        .withOnDemandAllocationStrategy(description.getOnDemandAllocationStrategy())
+                        .withOnDemandBaseCapacity(description.getOnDemandBaseCapacity())
+                        .withOnDemandPercentageAboveBaseCapacity(
+                            description.getOnDemandPercentageAboveBaseCapacity())
+                        .withSpotAllocationStrategy(description.getSpotAllocationStrategy())
+                        .withSpotInstancePools(description.getSpotInstancePools())
+                        .withSpotMaxPrice(description.getSpotPrice()));
+
+        updateReq.withMixedInstancesPolicy(mip);
+      } else {
+        updateReq.withLaunchTemplate(
+            new LaunchTemplateSpecification()
+                .withLaunchTemplateId(command.launchTemplateVersion.getLaunchTemplateId())
+                .withVersion(String.valueOf(command.launchTemplateVersion.getVersionNumber())));
+      }
+    }
+
```

Comment:
Would like to see this broken into smaller methods. Hard to follow with the nestings while checking and setting values.